src/pykeen/nn/node_piece.py

```diff
@@ -154,15 +157,18 @@ def extra_repr(self) -> Iterable[str]:
 
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}({', '.join(self.extra_repr())})"
-
+    
 
 class DegreeAnchorSelection(AnchorSelection):
     """Select entities according to their (undirected) degree."""
 
-    def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
+    def __call__(self, edge_index: numpy.ndarray, known_anchors: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
         unique, counts = numpy.unique(edge_index, return_counts=True)
-        top_ids = numpy.argpartition(counts, max(counts.size - self.num_anchors, 0))[-self.num_anchors :]
-        return unique[top_ids]
+        top_ids = numpy.argpartition(counts, max(counts.size - self.num_anchors, 0))
```

Comment:
[`argpartition`](https://numpy.org/doc/stable/reference/generated/numpy.argpartition.html) only guarantees that the `k=max(counts.size - self.num_anchors, 0)` element is in order and everything smaller is in front and everything larger behind. Since you afterwards may discard some of the first part, elements from the latter may be accessed. In this case, the elements may not be the largest available accordding to the degree.

The easiest fix I can see now which still maintains full vectorization without any loops it to fully arg-sort the array instead.

Reply:
good catch, changed to argsort + helper function in https://github.com/pykeen/pykeen/pull/776/commits/5d6f7aad471c9d9e131905d76f25b28f213f0d30